File: modules/coin-flip/ios/CoinFlipModule.swift
import ExpoModulesCore

public class CoinFlipModule: Module {
  public func definition() -> ModuleDefinition {
    Name("CoinFlip")

    Function("flipCoin") { () -> String in
      Int.random(in: 0...1) == 0 ? "heads" : "tails"
    }
  }
}

File: modules/coin-flip/android/src/main/java/expo/modules/coinflip/CoinFlipModule.kt
package expo.modules.coinflip

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import java.net.URL

class CoinFlipModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("CoinFlip")

    Function("flipCoin") {
      return@Function if ((0..1).random() == 0) "heads" else "tails"
    }
  }
}
